Improve clustering method documentation; make it skimmable (#4718)
OpenRefine/OpenRefine: docs/docs/manual/cellediting.md

### Clustering methods {#clustering-methods}

You don’t need to understand the details behind each clustering method to apply them successfully to your data.

The order in which these methods are presented in the interface and on this page is the order we recommend - starting with the most strict rules and moving to the most lax, which require more human supervision to apply correctly.

The clustering pop-up window offers you two categories of clustering methods: 6 different key collision clustering methods and 2 nearest neighbour clustering methods. 

> - [Key collision](#key-collision)

Reviewer: ```suggestion
- [Key Collision](#key-collision)
```

Could you capitalise every summary items? Like the title there are below. There is more than 10 that could be capitalized from what I saw.

**<a name="n-gram">N-gram fingerprinting</a>**

N-gram fingerprinting allows you to set the _n_ value to whatever number you’d like and will create n-grams of _n_ size (after doing some cleaning), alphabetize them, then join them back together into a  fingerprint.

**For example**, a 1-gram fingerprint will simply organize all the letters in the cell into alphabetical order - by creating segments one character in length. A 2-gram fingerprint will find all the two-character segments, remove duplicates, alphabetize them, and join them back together (for example, “banana” generates “ba an na an na,” which becomes “anbana”).

This can help match cells that have typos, or incorrect spaces (such as matching “lookout” and “look out,” which fingerprinting itself won’t identify because it separates words). The higher the _n_ value, the fewer clusters will be identified. With 1-grams, keep an eye out for mismatched values that are near-anagrams of each other (such as “Wellington” and “Elgin Town”).

For an in-depth understanding of N-gram fingerprinting, check this [document](https://github.com/OpenRefine/OpenRefine/wiki/Clustering-In-Depth#n-gram-fingerprint)

> ##### Phonetic clustering {#phonetic-clustering}

Reviewer: ```suggestion
_**Phonetic clustering {#phonetic-clustering}**_
```

Here, we have a complicated problem. The Header level 4 is much smaller than the paragraph title. So, could we put it the same size, but in italic? It's really an informative header…

### Clustering methods {#clustering-methods}

You don’t need to understand the details behind each clustering method to apply them successfully to your data.

The order in which these methods are presented in the interface and on this page is the order we recommend - starting with the most strict rules and moving to the most lax, which require more human supervision to apply correctly.

> The clustering pop-up window offers you a variety of clustering methods; 6 different key collision clustering methods and 2 nearest neighbour clustering methods.

Reviewer: ```suggestion
The clustering pop-up window offers you two categories of clustering methods: 6 different key collision clustering methods and 2 nearest neighbour clustering methods.
```
